homeassistant/components/purpleair/__init__.py

```diff
@@ -23,6 +25,26 @@ async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
 
     await hass.config_entries.async_forward_entry_setups(entry, PLATFORMS)
 
+    entry.async_on_unload(entry.add_update_listener(async_handle_entry_update))
+
+    return True
+
+
+async def async_handle_entry_update(hass: HomeAssistant, entry: ConfigEntry) -> None:
+    """Handle an options update."""
+    if entry.options.get(CONF_LAST_UPDATE_SENSOR_ADD) is True:
+        # If the last options update was to add a sensor, we reload the config entry:
+        await hass.config_entries.async_reload(entry.entry_id)
+
+
+async def async_remove_config_entry_device(
+    hass: HomeAssistant, config_entry: ConfigEntry, device_entry: dr.DeviceEntry
+) -> bool:
+    """Remove a config entry from a device."""
+    removed_sensor_index = async_get_sensor_index(hass, config_entry, device_entry)
+    new_entry_options = async_untrack_sensor_index(config_entry, removed_sensor_index)
+    hass.config_entries.async_update_entry(config_entry, options=new_entry_options)
```

Comment:
Is this used?

Reply:
This is so that removal of a device untracks the corresponding sensor. https://developers.home-assistant.io/docs/device_registry_index?_highlight=async_remove_#removing-devices

Reply:
Of course.  Guess I shouldn't have looked at this at the end of the day 😄